### PR TITLE
MFA for Admin Actions: Cert Authority

### PR DESCRIPTION
**Pull request by Joerger**
Require MFA for CA CRUD.

Part of [RFD 131](https://github.com/gravitational/teleport/blob/6b998a78ddd76c0c9eb7d966086f984b01144de7/rfd/0131-adminitrative-actions-mfa.md#administrative-actions).

### CODE REVIEW

File: lib/auth/trust/trustv1/service.go [hunk 1]
```diff
@@ -229,6 +238,8 @@ func (s *Service) GenerateHostCert(
 		return nil, trace.Wrap(err)
 	}
 
+	// TODO (Joerger): in v16.0.0, this endpoint should require admin action authorization.
```
Comment (Tener) [inline lib/auth/trust/trustv1/service.go:241]: Why do we have to wait for v16? What is the general plan?

File: tool/tctl/common/admin_action_test.go [hunk 2]
```diff
@@ -180,6 +183,58 @@ func (s *AdminActionTestSuite) TestRole() {
...
+	t := s.T()
+	ctx := context.Background()
+
+	priv, pub, err := native.GenerateKeyPair()
+	require.NoError(t, err)
+
+	key, cert, err := tlsca.GenerateSelfSignedCA(pkix.Name{CommonName: "Host"}, nil, time.Minute)
+	require.NoError(t, err)
+
+	ca, err := types.NewCertAuthority(types.CertAuthoritySpecV2{
```
Comment (Tener) [inline tool/tctl/common/admin_action_test.go:196]: `err` is unused according to the linter, which looks correct.

File: lib/auth/trust/trustv1/service.go (hunk 1 above)
Comment (Joerger) [reply lib/auth/trust/trustv1/service.go:241]: This endpoint was added in a minor version of v14, so to keep compatibility with v14.0 clients that still use the http endpoint, we can't require MFA until v16

Comment (Tener) [reply lib/auth/trust/trustv1/service.go]: Ah, ok, I see. Makes sense.

**Review by Tener — APPROVED** [review]

File: lib/auth/trust/trustv1/service.go (hunk 1 above)
Comment (espadolini) [reply lib/auth/trust/trustv1/service.go]: Couldn't we require MFA in the grpc endpoint but not in the http one? Or would that be too inconsistent?

**Review by espadolini — APPROVED** [review]

**Pull request merged**